src/devices/machine/spifi3.h

```diff
@@ -0,0 +1,465 @@
+// license:BSD-3-Clause
+// copyright-holders:Brice Onken,Tsubai Masanari,Olivier Galibert
+// thanks-to:Patrick Mackinlay
+
+/*
+ * HP 1TV3-0302 SPIFI3-SE SCSI controller
+ *
+ * Datasheets for this seem to be impossible to find - the only avaliable implementation to reference that I have
+ * found is the Sony NEWS APBus NetBSD driver. Hopefully a datasheet will turn up eventually.
+ * Based on internet research, it seems some HP PA-RISC systems also used the SPIFI3, including the E55.
+ *
+ * Because this driver was developed to work with NetBSD, NEWS-OS, and the NWS-5000 monitor ROM, only
+ * the features and flows that Sony used are implemented. Emulating non-Sony designs using this chip will likely
+ * require similar RE work to determine the exact SPIFI features used and add support for them into this driver.
+ * In its current state, this driver is unlikely to work out of the box with any other machines.
+ *
+ * Register definitions were derived from the NetBSD source code, copyright (c) 2000 Tsubai Masanari.
+ * SCSI state machine code was derived from the MAME NCR5390 driver, copyright (c) Olivier Galibert
+ *
+ * References:
+ * - https://github.com/NetBSD/src/blob/trunk/sys/arch/newsmips/apbus/spifireg.h
+ * - https://github.com/NetBSD/src/blob/trunk/sys/arch/newsmips/apbus/spifi.c
+ * - https://github.com/mamedev/mame/blob/master/src/devices/machine/ncr5390.cpp
+ */
+
+#ifndef MAME_MACHINE_SPIFI3_H
+#define MAME_MACHINE_SPIFI3_H
+
+#pragma once
+
+#include <queue>
+#include "machine/nscsi_bus.h"
```

Comment:
You should put the standard library headers after the the project headers, to avoid getting into a state where a header requires you to include another header before it.